### PR TITLE
Feature/user access management

### PR DESCRIPTION
**Pull request by ShadabFaiz**
#  Description
Implemented User Access Management

### Type of change

-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

## How Has This Been Tested?
- Manual Testing

## Checklist:

-   [x] My code follows the style guidelines of this project
-   [x] I have performed a self-review of my own code
-   [x] I have made corresponding changes to the documentation
-   [x] My changes generate no new warnings
-   [ ] I have added tests that prove my fix is effective or that my feature works
-   [x] New and existing unit tests pass locally with my changes
-   [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/utils/access/accessChecker.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { MODULE_CODES, MODULES } from "./modules.list";
+
+type actionType<T extends MODULE_CODES> = keyof typeof MODULES[T]["actionsAvailable"];
+
+/**
+ * @description Check whether loggedin user has access to the given module or not.
+ */
+export const UserHasAccess = <T extends MODULE_CODES>(moduleName: T, action: actionType<T>) => {
```
Comment (anirudh411) [inline src/utils/access/accessChecker.ts:8]: Why does the function name starts with a capital U?

Comment (ShadabFaiz) [reply src/utils/access/accessChecker.ts:8]: only variable names should start from lowercase, Classes / function should start from Uppercase to distinguish them.

Comment (anirudh411) [reply src/utils/access/accessChecker.ts:8]: A function name can be considered a variable. Never heard of functions starting with a capital letter, only classes or interfaces

Comment (ShadabFaiz) [reply src/utils/access/accessChecker.ts]: changed the methond name to camelCase

**Review by anirudh411 — APPROVED** [review]

**Pull request merged**